code-challenges.test.js

```diff
@@ -15,43 +15,144 @@
+//   })
+// })
+
+// const colors1 = ["purple", "blue", "green", "yellow", "pink"]
 // Expected output example (can be a different order): ["yellow", "blue", "pink", "green"]
-const colors2 = [
-  "chartreuse",
-  "indigo",
-  "periwinkle",
-  "ochre",
-  "aquamarine",
-  "saffron"
-]
+// const colors2 = [
+//   "chartreuse",
+//   "indigo",
+//   "periwinkle",
+//   "ochre",
+//   "aquamarine",
+//   "saffron"
+// ]
 // Expected output example (can be a different order): ["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]
+// FAIL  ./code-challenges.test.js
+// shuffleContent
+//   ✕ removes the first item from the array and shuffles the remaining content
+// ● shuffleContent › removes the first item from the array and shuffles the remaining content
+//   ReferenceError: shuffleContent is not defined
+
+// Pseudo code:
+// input: array
+// output: array with first item removed, then shuffles remaining items
+// process: 
+// iteration is needed, maybe for loop? or maybe .filter() ?
+// first remove first item of array, can use .shift() as it removes value at the first index.
+// Math.random() arrays index and not the value.
+
 
 // b) Create the function that makes the test pass.
 
-// Pseudo code:
+// const shuffleContent = (array) => {
+//   let newArr = []
+//   for ( let i=0; i < array.length; i++){
+//     newArr.shift(array[i].Math.random())
+//   }
+//   return newArr
+// }
+
+// console.log(shuffleContent(colors1, colors2))
 
 // --------------------2) Create a function that takes in an object that contains up votes and down votes and returns the end tally.
 
 // a) Create a test with expect statements for each of the variables provided.
 
-const votes1 = { upVotes: 13, downVotes: 2 }
+// describe("endTally", () => {
+//   it("takes in an object that contains up votes and down votes and returns the end tally", () => {
+//     const votes1 = { upVotes: 13, downVotes: 2 }
+//     const votes2 = { upVotes: 2, downVotes: 33 }
+//     expect(endTally(votes1)).toEqual("11")
+//     expect(endTally(votes2)).toEqual("-31")
+//   })
+// })
+
+// const votes1 = { upVotes: 13, downVotes: 2 }
 // Expected output: 11
-const votes2 = { upVotes: 2, downVotes: 33 }
+// const votes2 = { upVotes: 2, downVotes: 33 }
 // Expected output: -31
 
-// b) Create the function that makes the test pass.
+// FAIL  ./code-challenges.test.js
+// endTally
+//   ✕ takes in an object that contains up votes and down votes and returns the end tally (1 ms)
+// ● endTally › takes in an object that contains up votes and down votes and returns the end tally
+// ReferenceError: endTally is not defined
 
 // Pseudo code:
+// input: object, 2 key:value pairs
+// output: end tally
+// process: subtract the value of the keys
+        // dot notation can call on value
+        // can it subtract though?
+        // going deeper into classes.. maybe use a constructor
+        // am I overthinking this?
```

Comment:
Definitely overthinking! We know that the values of the objects are numbers, which means we can perform arithmetic with them. Using dot notation to access the values, we can just subtract the value of downvotes from the value of upvotes.